### DISCUSSION TITLE
Drop visits if travel time between two nodes exceeds

### DISCUSSION THREAD

**Discussion by despondency** [google/or-tools#2821] [Routing (and legacy CP) questions] [answered]
Is it possible to limit the solver to drop visits if time between two nodes is more than N time? Excluding this requirement for start location to first node.

**Comment by lperron** [comment, marked as answer]
Just tweak the distance matrix to return a very large value in that case.

**Comment by despondency** [comment]
But large number wont make the solver drop it? What constraint I need to add, to make the solver return it as dropped node? Again this needs to only happen for all node to node locations other than the depo to first node.

**Comment by lperron** [comment]
You can remove the destination from the next variable of the source.
But I am not sure it will be faster/
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le jeu. 7 oct. 2021 à 17:22, Mario Georgiev ***@***.***> a
écrit :

**Comment by despondency** [comment]
```
routing.nextVar(currentNode).delete();
```

This would make it so currentNode will be returned as dropped? 

Could you please follow up with a sample?

Thanks!

**Reply by Mizux** [reply]
```python
index = manager.NodeToIndex(node_to_delete)
routing.NextVar(current_index).RemoveValue(index)
```

**Comment by despondency** [comment]
This leads to no solution, if i try to isolate a node from all the other nodes. I want the solver to drop a certain node, if there is ANY other node that can reach it in time > some time. I want to receive it as a dropped node, rather than me actually dropping it completely from the nodes beforehand by checking all times between this node and all others. Is it possible?
